example/lib/ios_popover/_ios_popover_menu.dart

```diff
@@ -0,0 +1,222 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+class IosPopoverMenu extends StatelessWidget {
+  const IosPopoverMenu({
+    super.key,
+    this.radius = const Radius.circular(12),
+    this.arrowWidth = 18.0,
+    this.arrowLength = 12.0,
+    required this.arrowDirection,
+    this.backgroundColor = const Color(0xFF474747),
+    this.padding = EdgeInsets.zero,
+    this.arrowFocalPoint,
+    required this.child,
+  });
+
+  /// Radius of the decoration corners.
+  final Radius radius;
+
+  /// Extent of the arrow.
+  final double arrowLength;
+
+  /// Width of the arrow.
+  final double arrowWidth;
+
+  /// Direction where the arrow points to.
+  final ArrowDirection arrowDirection;
```

Comment:
I think that this approach to arrow positioning is going to require more calculation outside the popover than we want to do.

When we show a popover, we're going to have some target rectangle on the screen. We'll display the popover near that rectangle, and we want to the arrow to point towards that rectangle as much as possible. I think this concept of `arrowDirection` and `arrowFocalPoint` will require the user of this popover to run those position and direction calculations. I think that behavior makes more sense within the popover. The popover should be the one to decide where the arrow belongs, given a global `Offset` focal point.

Can you try that approach?

Reply:
PR updated with this approach.